addons/web_editor/static/src/js/backend/convert_inline.js

```diff
@@ -395,7 +395,7 @@ FieldHtml.include({
      * @override
      */
     commitChanges: function () {
-        if (!this.wysiwyg) {
+        if (this.mode == "readonly") {
```

Comment:
I would check that we are in edit mode, do the following 
`        if (this.nodeOptions['style-inline']) {
            this._toInline();	            this._toInline();
        }	        }
`

then returning `this._super();` in all cases